src/Examples/Examples/ResponseExtensions.cs

```diff
@@ -37,22 +44,35 @@ public static void MatchesExample(this IResponse response, string content, Func<
 			// the client encodes characters such as commas, so decode to compare
 			var decodedAbsolutePath = HttpUtility.UrlDecode(response.ApiCall.Uri.AbsolutePath);
 
-			decodedAbsolutePath.Should().Be(example.Uri.AbsolutePath.Length > 1
-				? example.Uri.AbsolutePath.TrimEnd('/')
-				: example.Uri.AbsolutePath);
+			var expectedUri = example.RequestUri.Uri;
+			var expectedPath = expectedUri.AbsolutePath.Length > 1 ? expectedUri.AbsolutePath.TrimEnd('/') : expectedUri.AbsolutePath;
+
+			// A lot of the examples are not constrained to an index which is not necessary what users typically do
+			// in NEST you have to be explicit and call AllIndices() which explicitly puts _all on the path
+			// So if we expect `/_search` but `/_all/_search` is passed we feel they are equivalent
+			if (expectedPath != "/_search" || decodedAbsolutePath != "/_all/_search")
+				decodedAbsolutePath.Should().Be(expectedPath);
 
 			// check expected query string params. Rather that _all_ keys match,
 			// only check that the ones in reference doc example are present, because
 			// the client may append more key/values such as "typed_keys"
-			var expectedQueryParams = HttpUtility.ParseQueryString(example.Uri.Query);
+			// Because the tests remove keys with string replace the following bad query string can appear
+			example.RequestUri.Query = example.RequestUri.Query.Replace("?&", "?").Replace("&&", "&");
+			var expectedQueryParams = HttpUtility.ParseQueryString(example.RequestUri.Query);
```

Comment:
`HttpUtility.ParseQueryString` cannot handle keys with no associated value, so should we handle this logic ourselves by splitting, `Uri.UnescapeDataString()` and creating a `Dictionary<string,string>` (which I think is OK for all params of Elasticsearch, as none a multi value that requires `NameValueCollection`)? It will also simplify the `foreach` loop below